src/inbox/mod_inbox_rdbms.erl

```diff
@@ -72,15 +72,19 @@ init(HostType, _Options) ->
     rdbms_queries:prepare_upsert(HostType, inbox_upsert, inbox,
                                  InsertFields,
                                  [<<"msg_id">>,
-                                  {expression, <<"box">>, <<"CASE WHEN inbox.box='archive' THEN ? ELSE inbox.box END">>},
+                                  {expression, <<"box">>, <<"CASE WHEN ?='bin' THEN 'bin'",
+                                                                " WHEN inbox.box='archive' THEN 'inbox'",
+                                                                " ELSE inbox.box END">>},
```

Comment:
Could you help me understand this logic? It seems like we can only set `inbox`, `bin` or leave the previous value unchanged, while previously we could set other boxes. What if moving to some other box is aggregated asynchronously? Am I missing something?

Reply:
The trick is that setting the box is still a synchronous request, I haven't added it to the list of aggregates, partially because is simpler that way, partially because I can assume setting the box to be an operation that doesn't happen in a tight loop. So aggregates are:
* only updating the inbox record, which will try to do inbox if it was archived and otherwise no box change
* resetting the count which does no box change
* removing the record, which does box change to the bin.
So at the end only moves to the bin, or moves from-archive-to-inbox, are attempted.

Reply:
To be a bit more pedantic, now that I think about it, one can also argue that the remove-record is also an infrequent operation and can stay synchronous. It also does not interfere with the others, once in the bin an automatic archive-to-inbox won't change anything. That would make the aggregation much simpler. Didn't think about it before...
Making set-inbox-properties async would also be an option, perhaps even better performance with one less operation being sync, but it would make the aggregation quite more complex. Still, I have a bazillion extensions in client code that make the aggregation several orders of magnitude more complex than properties and I managed well, so I can try adding these operations to the aggregation as well.